scripts/scil_validate_bids.py

```diff
@@ -337,131 +294,156 @@ def associate_dwis(layout, nSub):
+                              direction=directions[1],
+                              **base_dict) +\
+        layout.get(part='mag',
+                   PhaseEncodingDirection=phaseEncodingDirection[1],
+                   direction=directions[1],
+                   **base_dict)
+
+    all_associated_dwis = []
+    logging.info('Number of dwi: {}'.format(len(all_dwis)))
+    logging.info('Number of rev_dwi: {}'.format(len(all_rev_dwis)))
+    while len(all_dwis) > 0:
+        curr_dwi = all_dwis[0]
+
+        curr_association = [curr_dwi]
+        rev_curr_entity = curr_dwi.get_entities()
+
+        rev_iter_to_rm = []
+        for iter_rev, rev_dwi in enumerate(all_rev_dwis):
+            # At this stage, we need to check only direction
+            if 'direction' in curr_dwi.entities:
+                rev_curr_entity['direction'] = rev_curr_entity['direction'][::-1]
+                if rev_curr_entity == rev_dwi.get_entities():
+                    curr_association.append(rev_dwi)
+                    rev_iter_to_rm.append(iter_rev)
+            elif curr_dwi.entities['PhaseEncodingDirection'] == rev_dwi.entities['PhaseEncodingDirection'][:-1] and rev_curr_entity == rev_dwi.get_entities():
+                curr_association.append(rev_dwi)
+                rev_iter_to_rm.append(iter_rev)
+
+        # drop all rev_dwi used
+        logging.info('Checking dwi {}'.format(all_dwis[0]))
+        for item_to_remove in rev_iter_to_rm[::-1]:
+            logging.info('Removing item {} from rev_dwi'.format(item_to_remove))
+            del all_rev_dwis[item_to_remove]
+
+        # Add to associated list
+        if len(curr_association) < 3:
+            all_associated_dwis.append(curr_association)
+        else:
+            logging.info("These acquisitions have too many associated dwis.")
+        del all_dwis[0]
+
+    if len(all_rev_dwis):
+        for curr_rev_dwi in all_rev_dwis:
+            all_associated_dwis.append([curr_rev_dwi])
 
-    return all_dwis
+    return all_associated_dwis
 
 
 def main():
     parser = _build_arg_parser()
     args = parser.parse_args()
 
+    assert_inputs_exist(parser, [], args.bids_ignore)
     assert_outputs_exist(parser, args, args.out_json)
 
     log_level = logging.INFO if args.verbose else logging.WARNING
     logging.getLogger().setLevel(log_level)
     coloredlogs.install(level=log_level)
 
     data = []
-    layout = BIDSLayout(args.in_bids, validate=False,
-                        ignore=_load_bidsignore_(args.in_bids))
-    subjects = layout.get_subjects()
-
-    if args.participants_label:
-        subjects = [nSub for nSub in args.participants_label if nSub in subjects]
+    bids_indexer = BIDSLayoutIndexer(validate=False,
+                                     ignore=_load_bidsignore_(args.in_bids,
+                                                              args.bids_ignore))
+    layout = BIDSLayout(args.in_bids, indexer=bids_indexer)
```

Comment:
Do you want to enforce the presence of the `dataset_description.json` file ? If not, add the `validate=False` parameter back here